MonoGame.Framework/Graphics/SpriteBatch.cs

```diff
@@ -126,30 +128,36 @@ void Setup()
 			gd.RasterizerState = _rasterizerState;
 			gd.SamplerStates[0] = _samplerState;
 			
-            // Setup the default sprite effect.
 			var vp = gd.Viewport;
+            if ((vp.Width != _lastViewport.Width) || (vp.Height != _lastViewport.Height))
+            {
+                // Normal 3D cameras look into the -z direction (z = 1 is in font of z = 0). The
+                // sprite batch layer depth is the opposite (z = 0 is in front of z = 1).
+                // --> We get the correct matrix with near plane 0 and far plane -1.
+                Matrix.CreateOrthographicOffCenter(0, vp.Width, vp.Height, 0, 0, -1, out _projection);
 
-		    Matrix projection;
-
-            // Normal 3D cameras look into the -z direction (z = 1 is in font of z = 0). The
-            // sprite batch layer depth is the opposite (z = 0 is in front of z = 1).
-            // --> We get the correct matrix with near plane 0 and far plane -1.
-            Matrix.CreateOrthographicOffCenter(0, vp.Width, vp.Height, 0, 0, -1, out projection);
+                // Some platforms require a half pixel offset to match DX.
+                if (NeedsHalfPixelOffset)
+                {
+                    _projection.M41 += -0.5f * _projection.M11;
+                    _projection.M42 += -0.5f * _projection.M22;
+                }
 
-            // Some platforms require a half pixel offset to match DX.
-            if (NeedsHalfPixelOffset)
-            {
-                projection.M41 += -0.5f * projection.M11;
-                projection.M42 += -0.5f * projection.M22;
+                _lastViewport = vp;
             }
 
             if (_matrix.HasValue)
             {
                 var transformMatrix = _matrix.GetValueOrDefault();
-                Matrix.Multiply(ref transformMatrix, ref projection, out projection);
+                Matrix projection;
+                Matrix.Multiply(ref transformMatrix, ref _projection, out projection);
```

Comment:
Nuh, it's still slower.
But assuming that most value of the _projection/OrthographicOffCenter are either 0 or 1, we can unroll it too and remove redundant operations.